devices.py

```diff
@@ -59,35 +61,296 @@ class Newport(EpicsMotor):
     def move(self, *args, **kwargs):
         return super().move(*args, **kwargs)
 
 
-class SeqBase(SndDevice):
+class Sequencer(Device):
     """
     Base sequencer class.
+    
+    Attributes
+    ----------
+    state_control : Component
+        start and stop the sequencer with a binary signal. Use Start/Stop
+        methods.
+
+    set_run_count_pattern : Component
+        Epics enum for defining whether to run the sequence once, n times or
+        forever.
+    
+    set_run_count : Component
+        If running n times, set n.
+
+    sequence_owner : Component
+        Numeric sequence owner
+
+    sequence_owner_name : Component
+        Enum sequence owner
+
+    photon_beam_owner : Component
+        Numberic beam owner
+
+    photon_beam_owner_name : Component
+        Enum beam owner
+    
     """
-    state_control = Cmp(EpicsSignal, ":PLYCTL")
+
+    # left-most column
+    state_control = Cpt(EpicsSignal, ":PLYCTL")
+    set_run_count_pattern = Cpt(EpicsSignal, ":PLYMOD")
+    set_run_count = Cpt(EpicsSignal, ":REPCNT")
+
+    # top row
+    sequence_owner = Cpt(EpicsSignal, ":HUTCH_ID")
+    sequence_owner_name = Cpt(EpicsSignal, ":HUTCH_NAME")
+    photon_beam_owner = FCpt(
+        EpicsSignal, "{self._beam_owner_prefix}:BEAM_OWNER_ID")
+    photon_beam_owner_name = FCpt(
+        EpicsSignal, "{self._beam_owner_prefix}:BEAM_OWNER_NAME")
     
+    # center column
+    beam_rate = FCpt(EpicsSignalRO, "{self._beam_rate_pv}")
+    play_count = Cpt(EpicsSignalRO, ":PLYCNT")
+    total_play_count = Cpt(EpicsSignalRO, ":TPLCNT")
+    play_status = Cpt(EpicsSignalRO, ":PLSTAT")
+    current_step = Cpt(EpicsSignalRO, ":CURSTP")
+
+    # right column
+    sync_marker = Cpt(EpicsSignal,":SYNCMARKER")
+    next_sync = Cpt(EpicsSignal,":SYNCNEXTTICK")
+    run_using = Cpt(EpicsSignal,":BEAMPULSEREQ")
+
+    # add some properties 
+    # learn about hints methods 
+
     def __init__(self, prefix, timeout=1, *args, **kwargs):
+        self._beam_rate_pv = "EVNT:SYS0:1:LCLSBEAMRATE"
+        self._beam_owner_prefix = "ECS:SYS0:0"
         super().__init__(prefix, *args, **kwargs)
         self.timeout = timeout
 
-    def start(self):
+    def start(self,wait=True):
         """
         Start the sequencer.
         """
         status = self.state_control.set(1, timeout=self.timeout)
-        status_wait(status)
+        if wait:
+            status_wait(status)
         
-    def stop(self):
+    def stop(self,wait=True):
         """
         Stop the sequencer.
         """
         status = self.state_control.set(0, timeout=self.timeout)
-        status_wait(status)        
+        if wait:
+            status_wait(status)        
+
+
+class ConsolidatedSamplePalette(Device):
+    '''
+    
+    '''
+    # x_motor = Cpt(IMS,"{self._xmotor_prefix}:{self._xmotor_suffix}")
+    # y_motor = Cpt(IMS,"{self._ymotor_prefix}:{self._ymotor_suffix}")
+    # z_motor = Cpt(IMS,"{self._zmotor_prefix}:{self._zmotor_suffix}")
```

Comment:
I know these were commented out, but would have needed to be `FCpt`. Speaking of, one thing I've noticed with `FCpt` is that it takes a good amount longer to instantiate than a standard `Cpt`, so just keep that in mind when writing these in the future.